dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py

```diff
@@ -31,9 +31,9 @@ def process_ibex_sim_log(ibex_log, csv):
             if re.search("ecall", line):
               break
             # Extract instruction infromation
-            m = re.search(r"^\s*(?P<time>\d+)\s+(?P<cycle>\d+) " \
-                          "(?P<pc>[0-9a-f]+) (?P<bin>[0-9a-f]+) (?P<instr>.*)" \
-                          "x(?P<rd>\d+)=0x(?P<val>[0-9a-f]+)", line)
+            m = re.search(r"^\s*(?P<time>\d+)\s+(?P<cycle>\d+)\s+" \
+                          "(?P<pc>[0-9a-f]+)\s+(?P<bin>[0-9a-f]+)\s+(?P<instr>.*)" \
+                          ".*x(?P<rd>\d+)=0x(?P<val>[0-9a-f]+)", line)
```

Comment:
Now that the entries are tab-separated, you can use `line.strip().split("\t")`

That's code I've been using:


```python
def get_instructions_from_trace(trace_file):
    instructions = {}
    with open(trace_file) as f:
        line_cnt = -1
        for line in f:
            line_cnt += 1
            if line_cnt == 0:
                continue
            elements = line.strip().split("\t")
            time = elements[0]
            cycle = elements[1]
            pc = int(elements[2], base=16)
            instr = int(elements[3], base=16)
            opcode = elements[4]
            desc = ''
            if len(elements) > 5:
                desc = elements[5]
            instructions[pc] = {'instr': instr, 'opcode': opcode, 'desc': desc, 'pc': pc}
    return instructions
```

Reply:
I think keeping our regex is better for our post-simulation comparison flows, as we only really care about the RD register and the rd_wdata, I think the regex allows us to extract that information more easily for our use-case

Reply:
Sure, whatever works for your use case. (If you want more structured data, a separate tracer that makes use of the RVFI interface might be more future-proof, but that's not for now.)